src/Proxy/ProxyGenerator.php

```diff
@@ -940,7 +941,12 @@ private function generateMethods(ClassMetadata $class)
             if ($this->isShortIdentifierGetter($method, $class)) {
                 $identifier = lcfirst(substr($name, 3));
                 $fieldType  = $class->getTypeOfField($identifier);
-                $cast       = in_array($fieldType, ['integer', 'smallint'], true) ? '(int) ' : '';
+
+                if (ClassUtils::containsEnumType($method->getReturnType())) {
+                    $cast = '';
+                } else {
+                    $cast = in_array($fieldType, ['integer', 'smallint'], true) ? '(int) ' : '';
```

Comment:
Or better yet: idea that needs somebody with a deeper ORM knowledge than me to be validated: what if we get rid of the cast instead? The proxied entity will have the field already casted to a correct type thanks to initial id hydration. This `$cast` construct here is 10 years old and removing it doesn't break any test so I think it's worth challenging.

Reply:
Yes I think that's the best way.  

If the method is typed, for the scalar types PHP already cast return value if needed?

Reply:
@malarzm `AbstractProxyFactory::getProxy` is responsible for setting the id field on the proxy, its not gone through hydration, instead it might just be the value passed to `EntityManager::getReference($entity, $id)` without casting. So getting rid of the cast would require to move the cast to a higher level in the ORM. Same for UnitOfWork::createEntity using `getProxy`, associated ids don't have types and are not converted to int for example.

A test in ORM would be:

```php
$proxy = $em->getReference(CmsUser, "1");
$proxy->getId();
```
Unsure if we test this at the moment.

Reply:
Thanks @beberlei I'll try to have a closer look there in ORM 👍

Reply:
@malarzm any news on this?

Reply:
Quoting myself from #998 

> Probably we won't end with this exact solution as it seems ORM has no tests for this behaviour either. The change looks OK ORM-wise, but I'm afraid it may hit users in an unexpected and hard to debug way.

Sadly I didn't have time to get back to this :(

Reply:
How can I help ?